src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -113,103 +114,84 @@ def compilation_hash(otf_closure: stages.ProgramCall) -> int:
+                    otf_workflow=workflow.CachedStep(o.otf_workflow, hash_function=o.hash_function),
+                    name=o.name,
+                )
+            ),
+            cached_name="_cached",
+        )
+        device_type = core_defs.DeviceType.CPU
+        hash_function = compilation_hash
+        otf_workflow = factory.SubFactory(
+            GTFNCompileWorkflowFactory, device_type=factory.SelfAttribute("..device_type")
+        )
+        name = factory.LazyAttribute(lambda o: f"run_gtfn_{o.device_name}{o.cached_name}")
 
-GTFN_GPU_WORKFLOW = recipes.OTFCompileWorkflow(
-    translation=GTFN_GPU_TRANSLATION_STEP,
-    bindings=nanobind.bind_source,
-    compilation=GTFN_DEFAULT_COMPILE_STEP,
-    decoration=functools.partial(convert_args, device=core_defs.DeviceType.CUDA),
-)
-
+    executor = factory.LazyAttribute(
+        lambda o: otf_compile_executor.OTFCompileExecutor(otf_workflow=o.otf_workflow, name=o.name)
+    )
+    allocator = next_allocators.StandardCPUFieldBufferAllocator()
 
-gtfn_executor = otf_compile_executor.OTFCompileExecutor(
-    name="run_gtfn", otf_workflow=GTFN_DEFAULT_WORKFLOW
-)
-run_gtfn = otf_compile_executor.OTFBackend(
-    executor=gtfn_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
-)
 
-gtfn_imperative_executor = otf_compile_executor.OTFCompileExecutor(
-    name="run_gtfn_imperative",
-    otf_workflow=gtfn_executor.otf_workflow.replace(
-        translation=gtfn_executor.otf_workflow.translation.replace(use_imperative_backend=True),
-    ),
-)
-run_gtfn_imperative = otf_compile_executor.OTFBackend(
-    executor=gtfn_imperative_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
-)
+__config = config.get_config()
 
-# TODO(ricoh): add API for converting an executor to a cached version of itself and vice versa
-gtfn_cached_executor = otf_compile_executor.CachedOTFCompileExecutor(
-    name="run_gtfn_cached",
-    otf_workflow=workflow.CachedStep(
-        step=gtfn_executor.otf_workflow, hash_function=compilation_hash
-    ),
-)
-run_gtfn_cached = otf_compile_executor.OTFBackend(
-    executor=gtfn_cached_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
-)
+__user_defaults = {
+    "gpu": __config.get("use_gpu", False),
+    "cached": __config.get("cache_backends", False),
+}
 
+run_gtfn = GTFNBackendFactory(**__user_defaults)
 
-run_gtfn_with_temporaries = otf_compile_executor.OTFBackend(
-    executor=otf_compile_executor.OTFCompileExecutor(
-        name="run_gtfn_with_temporaries",
-        otf_workflow=gtfn_executor.otf_workflow.replace(
-            translation=gtfn_executor.otf_workflow.translation.replace(
-                lift_mode=LiftMode.FORCE_TEMPORARIES
-            ),
-        ),
-    ),
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
+run_gtfn_imperative = GTFNBackendFactory(
+    otf_workflow__translation__use_imperative_backend=True, **__user_defaults
```

Comment:
Is there a way to shorten these names? (`otf_workflow__translation__use_imperative_backend`)

Reply:
It's not a name, it's a path equivalent to `otf_workflow.translation.use_imperative_backend`.  However, that would be invalid python syntax in a keyword arg, so the authors of factoryboy went with `__` instead.

Reply:
The answer is still yes though, we can shorten them in two ways:
- make the individual attribute names shorter
- provide a parameter (under `class Params:`) in the parent factory with a shorter name and pass that on to the subfactory. (The full path attribute would still be available though.)